miniapps/toys/snake.cpp

```diff
@@ -309,6 +309,7 @@ int main(int argc, char *argv[])
    }
 
    // Clean up and exit
+   delete fec;
```

Comment:
How about this instead?
```diff
diff --git a/miniapps/toys/snake.cpp b/miniapps/toys/snake.cpp
index 77a8d2b52..f82bda41d 100644
--- a/miniapps/toys/snake.cpp
+++ b/miniapps/toys/snake.cpp
@@ -233,8 +233,8 @@ int main(int argc, char *argv[])
    mesh.FinalizeTopology();
 
    // Paint elements with alternating colors
-   FiniteElementCollection *fec = new L2_FECollection(0, 3, 1);
-   FiniteElementSpace fespace(&mesh, fec);
+   L2_FECollection fec(0, 3, 1);
+   FiniteElementSpace fespace(&mesh, &fec);
    GridFunction color(&fespace);
 
    for (int i=0; i<24; i++) { color[i] = (i%2)?1.0:-1.0; }
```

Reply:
`done` in https://github.com/mfem/mfem/pull/3582/commits/632e345f79ea1d7d12c70104f2472ee254d55a30